osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -26,11 +27,26 @@ public double BPMAt(double time)
             return 60000 / BeatLengthAt(time);
         }
 
-        public double BeatLengthAt(double time)
+        public double BPMMultiplierAt(double time)
         {
             ControlPoint overridePoint;
             ControlPoint timingPoint = TimingPointAt(time, out overridePoint);
-            return timingPoint.BeatLength;
+
+            return overridePoint?.VelocityAdjustment ?? 1;
+        }
+
+        public double BeatLengthAt(double time, bool addVelocity = true)

```

Comment:
My point was more leaning towards the second suggestion - that the parameter should be removed altogether and velocity multiplied implicitly. But yes you're right the first suggestion does not affect your proposal.